staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -188,6 +195,7 @@ func (p *restrictedProfile) Apply(pod *corev1.Pod, containerName string, target
 
 	switch style {
 	case podCopy:
+		removeLabelsAndProbes(pod)
```

Comment:
I agree with that but I think we need to manage this with on/off flag as we did in https://github.com/openshift/oc/blob/cf7b09cd29f1c0f56b2c6cd8193092c57c7730ff/pkg/cli/debug/debug.go#L252-L255 and default would be to removal of probes. WDYT? @verb

Reply:
@ardaguclu 

Thank you for your comment and I agree that.
`oc debug` seems to be implemented to be able to control the removal of probes, labels, annotations and initContainers.
On the other hands in `kubectl debug`, labels will be removed in all profiles and probes will be removed in some profiles(general and baseline).

So I'm wondering which should be targeted for removal control（only probes, probes and labels or all?）.
How do you think?

Current implementations of `kubectl debug` are as below.

#### probes
`removeLabelsAndProbes()` which remove probes already has been implemented and used in some profiles(general and baseline).
https://github.com/kubernetes/kubernetes/blob/91ee30074bee617d52fc24dc85132fe948aa5153/staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go#L245-L254

#### labels
`removeLabelsAndProbes()` also implemented to remove labels, but copy pod labels will be removed when copy pod is created before  `removeLabelsAndProbes()` will be called.
https://github.com/kubernetes/kubernetes/blob/91ee30074bee617d52fc24dc85132fe948aa5153/staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go#L580-L589

#### annotations and initContainers
There seems to be not implemented the logic to remove annotations and initContainers.

Reply:
For now, I've added `--keep-*` flags to be able to control the removal of probes, labels, annotations and initContainers like `oc debug`.